components/feature/toolbar/src/main/java/mozilla/components/feature/toolbar/ToolbarPresenter.kt

```diff
@@ -64,16 +65,39 @@ class ToolbarPresenter(
         val session = sessionId?.let { sessionManager.findSessionById(sessionId) }
             ?: sessionManager.selectedSession
 
-        renderer.post(session?.url ?: "")
+        val urlToPost = if (session != null) {
+            getURLToPost(session, session.url)
+        } else {
+            ""
+        }
+
+        // Set the title to be the url as we don't have a title until the page loads
+        session?.let { onTitleChanged(it, it.url) }
+
+        renderer.post(urlToPost)
         toolbar.displayProgress(session?.progress ?: 0)
         updateToolbarSecurity(session?.securityInfo ?: Session.SecurityInfo())
     }
 
     override fun onUrlChanged(session: Session, url: String) {
-        renderer.post(url)
+        renderer.post(getURLToPost(session, url))
         toolbar.setSearchTerms(session.searchTerms)
     }
 
+    private fun getURLToPost(session: Session, url: String): String {
+        return if (session.isCustomTabSession()) {
```

Comment:
I'm not a fan of adding custom tab checks around the codebase since it increases our complexity in multiple components at once. How about always passing the url in, and when `displayTitle` is enabled, parse the url for the host in the CustomTabsToolbarFeature itself which does the global "isCustomTab" check in one place.

Reply:
If I go with [Sebastian's recommendation here](https://github.com/mozilla-mobile/android-components/pull/2625#discussion_r272075518) how should I handle this? I agree that constantly checking for `isCustomTab` could be annoying.

Reply:
This can still be done within the `CustomTabsToolbarFeature` using a session observer to check when the title is received in that session.

The [`WindowFeature`](https://github.com/mozilla-mobile/android-components/blob/master/components/feature/session/src/main/java/mozilla/components/feature/session/WindowFeature.kt#L19-L34) has a nice example of how to make an internal session observer in a feature class. However, since you want to focus on just one session, you wouldn't want to inherit from `SelectionAwareSessionObserver`, but provide the `sessionId` of the CT to it and make changes.

There might also be the case where the title is already received so you would have to check the `session.title` before deciding to not show the title entirely.

Reply:
I think my plan is to do something like mentioned in my discussion with Sebastian [below](https://github.com/mozilla-mobile/android-components/pull/2625#discussion_r272077771). Does that sound good to you? Just not sure what solution will work with the config being the same between the two bars.